lit_nlp/examples/dalle_demo.py

```diff
@@ -0,0 +1,97 @@
+r"""Example dalle demo loading a custom model.
+
+To run locally with a small number of examples:
+  python -m lit_nlp.examples.dalle_demo \
+      --alsologtostderr --port=5432
+
+Then navigate to localhost:5432 to access the demo UI.
+"""
+import os
+import sys
+from typing import Optional, Sequence
+
+from absl import app
+from absl import flags
+from absl import logging
+
+from lit_nlp import dev_server
+from lit_nlp import server_flags
+from lit_nlp.api import layout
+from lit_nlp.examples.datasets import dalle_prompt
+from lit_nlp.examples.models import dalle
+
+# NOTE: additional flags defined in server_flags.py
+
+FLAGS = flags.FLAGS
+
+FLAGS.set_default("development_demo", True)
+
+_MODELS = flags.DEFINE_list(
+    "models", ["dalle-mini/dalle-mini/mega-1-fp16:latest", 'dalle-mini/dalle-mini/mini-1:v0'],
+    "Models to load")
+
+_MAX_EXAMPLES = flags.DEFINE_integer(
+    "max_examples", 10,
+    "Maximum number of examples to load from each evaluation set. Set to None to load the full set."
+)
+
+# Custom frontend layout; see api/layout.py
+modules = layout.LitModuleName
+Dalle_LAYOUT = layout.LitCanonicalLayout(
+    upper={
+        "Main": [
+            modules.EmbeddingsModule,
```

Comment:
I like the idea of supporting embeddings in this demo. Would be really cool if we had embeddings for the prompts and the images so that we could toggle between these fields, but that sounds somewhat complicated since the output will be multiple images and each will require their own embedding.